src/main/java/karlett/task/Deadline.java

```diff
@@ -0,0 +1,40 @@
+package karlett.task;
+
+import java.io.FileWriter;
+import java.io.IOException;
+
+public class Deadline extends Task {
+
+    protected String by;
+
+    /* constructor used for user input */
+    public Deadline(String description, String by) throws IOException {
+        this.description = description;
+        this.isDone = false;
+        this.by = by;
+        increaseNumberOfTasks();
+        printNewTaskAddedMessage();
+        appendNewTaskToFile();
+    }
+
+    /* constructor used for loading file data */
+    public Deadline(String description, String by, boolean isDone) throws IOException {
+        this.description = description;
+        this.isDone = isDone;
+        this.by = by;
+        increaseNumberOfTasks();
+    }
+
+    //@Override
```

Comment:
This comment does not explain much. Instead you could provide [javadoc comments](https://se-education.org/guides/conventions/java/basic.html#comments) to describe what this method is doing.